lib/updatehub/updatehub.c

```diff
@@ -220,7 +220,7 @@ static void cleanup_connection(void)
 		LOG_ERR("Could not close the socket");
 	}
 
-	memset(&ctx.fds[1], 0, sizeof(ctx.fds[1]));
+	memset(&ctx.fds[0], 0, sizeof(ctx.fds[0]));
```

Comment:
The case that you are fixing is not completely clear for me because this function is called from different contexts, but wouldn't be better (and more generic) if you iterate over the file descriptors and clean then ? e.g:

```
for (i = 0; i < ctx.nfds; i++) {
    memset(&ctx.fds[i], 0, sizeof(ctx.fds[i]));
}
```

Reply:
Sure.